apps/container/src/components/SearchBar.vue

```diff
@@ -0,0 +1,172 @@
+        class="item"
+      >
+        <v-icon
+          v-if="item.type"
+          :icon="item.type === 'teacher' ? 'mdi-account' : 'mdi-book'"
+          class="mr-3"
+        />
+        {{ item.name }}
+      </v-list-item>
+    </v-list>
+  </div>
+</template>
+
+<style scoped lang="scss">
+.wrapper {
+  position: relative;
+  width: 100%;
+}
+
+.wrapper:focus-within .results {
+  display: block;
+}
+
+.results {
+  position: absolute;
+  width: 100%;
+  max-height: 320px;
+  overflow-y: auto;
+  border-radius: 4px;
+  display: none;
+  z-index: 9999;
+}
+</style>
+
+<script lang="ts" setup>
+import { useQuery } from '@tanstack/vue-query';
+import debounce from 'lodash.debounce';
+import { Reviews } from 'services';
+import { computed, onMounted, ref, watch } from 'vue';
+import FeedbackAlert from './FeedbackAlert.vue';
+import router from '@/router';
+
+const searchTerm = ref('');
+
+const clear = () => {
+  searchTerm.value = '';
+  router.replace({
+    name: 'reviews',
+  });
+};
+
+onMounted(() => {
+  searchTerm.value = router.currentRoute.value.query.q as string;
+});
+
+watch(
+  () => router.currentRoute.value.query.q as string,
+  (q) => {
+    searchTerm.value = q;
+  },
+);
+
+const enterSearch = (id: string, type: string, name: string) => {
+  (document.activeElement as HTMLDivElement)?.blur();
+
+  searchTerm.value = name;
+  router.replace({
+    name: 'reviews',
+    query: {
+      q: name,
+      teacherId: type === 'teacher' ? id : undefined,
+      subjectId: type === 'subject' ? id : undefined,
+    },
+  });
+};
+
+const {
+  isError: isErrorTeachers,
+  isFetching: isFetchingTeachers,
+  data: searchResultsTeachers,
+  refetch: refetchTeachers,
+} = useQuery({
+  refetchOnWindowFocus: false,
+  queryKey: [
+    'reviews',
+    'search',
+    router.currentRoute.value.query.q as string,
+    'teachers',
+  ],
+  queryFn: () =>
+    Reviews.searchTeachers(router.currentRoute.value.query.q as string),
+  enabled: !!(router.currentRoute.value.query.q as string),
+});
+
+const {
+  isError: isErrorSubjects,
+  isFetching: isFetchingSubjects,
+  data: searchResultsSubjects,
+  refetch: refetchSubjects,
+} = useQuery({
+  refetchOnWindowFocus: false,
+  queryKey: [
+    'reviews',
+    'search',
+    router.currentRoute.value.query.q as string,
+    'subjects',
+  ],
+  queryFn: () =>
+    Reviews.searchSubjects(router.currentRoute.value.query.q as string),
+  enabled: !!(router.currentRoute.value.query.q as string),
+});
+
+const useSearch = debounce(() => {
+  refetchTeachers();
+  refetchSubjects();
+}, 500);
+
+const search = (e: InputEvent) => {
+  router.replace({
+    name: 'reviews',
+    query: {
+      q: (e.target as HTMLInputElement)?.value,
+    },
+  });
+  useSearch();
+};
+
+const processedResults = computed(() => [
+  ...(searchResultsTeachers.value?.data.data.map((result) => ({
+    name: result?.name,
+    id: result._id,
+    type: result._id && 'teacher',
```

Comment:
se result._id for falsy, ele viria com que valor?

Reply:
talvez de pra criar uma função que devolve esse map que foi feito de forma duplicada